### ISSUE TITLE
Check for use of semi-colons used to define multiple resources

### ISSUE THREAD

**Issue by alexjfisher** [rodjek/puppet-lint#547]
> Semicolons must not be used to declare multiple resources within a set of curly braces.
https://docs.puppet.com/guides/style_guide.html#resource-arrangement

eg
```puppet
file {
  '/tmp/file1':
    ensure => 'file',
    mode   => '0600';
  '/tmp/file2':
    ensure => 'file',
    mode   => '0644';
  '/tmp/dir':
    ensure => 'directory';
}

It'd be nice if puppet-lint could detect this.

**Comment by rnelson0** [comment]
Hrm, the good and bad at that link are the same. Regardless, I believe this guidance is out of date, as this is required for the new 'default' method. I will check with Puppet to see if it is correct before proceeding either way.

**Comment by alexjfisher** [comment]
The good/bad at that link are for the 

> resources should be grouped by logical relationship to each other

bit.  They don't have good/bad examples for 

> Semicolons must not be used to declare multiple resources within a set of curly braces.

so I added an illustration here.

**Comment by alexjfisher** [comment]
You don't actually come across many people using this style, but I did in `puppet-mrepo` https://github.com/voxpupuli/puppet-mrepo/commit/0c86b3a1b9e0be652adc74f81786385b9bc68de3

**Comment by ghoneycutt** [comment]
This spawned from me coding like the example @alexjfisher gave and no one liking it.

Good:

```puppet
file { '/tmp/file1':
  ensure => 'file',
  mode   => '0600',
}

file { '/tmp/file2':
  ensure => 'file',
  mode   => '0644',
}

file { '/tmp/dir':
  ensure => 'directory',
}
```

Bad

```puppet
file {
  '/tmp/file1':
    ensure => 'file',
    mode   => '0600';
  '/tmp/file2':
    ensure => 'file',
    mode   => '0644';
  '/tmp/dir':
    ensure => 'directory';
}
```

**Comment by ghoneycutt** [comment]
@rnelson0 what's the new default method you are speaking about? Could you please post a link?

**Comment by rnelson0** [comment]
See https://docs.puppet.com/puppet/4.6/reference/lang_resources_advanced.html#multiple-resource-bodies.

**Comment by ghoneycutt** [comment]
Given "Semicolons must not be used to declare multiple resources within a set of curly braces." in the current style guide, we should have a rule that enforces it.

**Comment by rnelson0** [comment]
I opened https://tickets.puppetlabs.com/browse/DOCUMENT-584 to track this concern.

**Comment by hlindberg** [comment]
I commented on DOCUMENT-584. The gist of that post is that the rule should be "semicolon separated multiple resource bodies should only be used in conjunction with a local default body"
